### PR TITLE
add `mint transaction` support

### LINKED ISSUE #702: Add support for the new mint transaction
The client's `Transaction` now has a `Mint` type, used to mint new coins. The SDK should have a helper interface to help users mint coins. Where this interface will live is up for discussion and is part of resolving this issue.

### PR DESCRIPTION
**Pull request by Salka1988**
closes #702 

add `mint transaction` support in

### CODE REVIEW

**Comment by Salka1988** [comment]
Is there or will there be some kind of endpoint on the `fuel-client` through which I will be able to do `mint transactions`
@digorithm @Voxelot

**Comment by digorithm** [comment]
> Is there or will there be some kind of endpoint on the `fuel-client` through which I will be able to do `mint transactions` @digorithm @Voxelot

I'm not sure I follow your question. Why would we need an _endpoint_ specifically for that? Doesn't sending a `Transaction` of type `Mint` already work for sending a transaction of type mint?

File: packages/fuels-signers/src/provider.rs [hunk 1]
```diff
@@ -141,6 +142,24 @@ impl Provider {
         }
     }
 
+    pub async fn mint_transaction(
```
Comment (digorithm) [inline packages/fuels-signers/src/provider.rs:145]: `mint_transaction` sounds like a verb. i.e., we're minting a transaction. Which is wrong. 

I'd suggest renaming it to `mint` only. Plus, add some doc-level comments here.

File: packages/fuels/tests/contracts.rs [hunk 2]
```diff
@@ -770,3 +771,26 @@ async fn test_contract_call_with_non_default_max_input() -> Result<(), Error> {
...
+        1,
+        AssetId::from([3u8; 32]),
+    )];
+
+    let a = provider.mint_transaction(output_coins).await?;
+
+    dbg!(&a);
+
+    dbg!(wallet.get_balances().await?);
+
```
Comment (digorithm) [inline packages/fuels/tests/contracts.rs:794]: Remove `dbg!` statements.

**Review by digorithm — CHANGES_REQUESTED** [review]
A couple of nits.

Also, it seems like mint transactions aren't supported by the client. Just noticed that: https://github.com/FuelLabs/fuel-core/blob/b50f0a9ec8a924ad6cf2e3cb12e4699d70bbb9e1/fuel-txpool/src/txpool.rs#L76. 

@Voxelot can you confirm that that's the case?

**Comment by Salka1988** [comment]
We’re not prioritizing this now and it needs to be better specified.

**Pull request closed without merging**